### PR TITLE
Better feature detection for assimp version

### PR DESCRIPTION
**Pull request by cottsay**
The unified headers were introduced in Assimp 2.0.1150, so checking for Assimp 3.0.0 is not quite the best solution.

See https://github.com/assimp/assimp/commit/6fa251c2f2e7a142bb861227dce0c26362927fbc

Tested on Fedora 19 (Assimp 2.0.1071) and Fedora 20 (Assimp 3.0.1264).

Should probably be tested on Ubuntu before release, but I don't foresee any issues.

Also, a backport to groovy-devel would be nice!

### CODE REVIEW

**Comment by cottsay** [comment]
Since I removed those #if IS_ASSIMP3 from the middle of collada_to_urdf.cpp, I'm not as confident that this will work completely. Why were those there to begin with? Was there an issue with export.h in assimp 2 that I'm not aware of?

File: collada_urdf/src/collada_to_urdf.cpp [hunk 1]
```diff
@@ -597,9 +593,7 @@ int main(int argc, char** argv)
     cerr << ";; Adding gazebo description" << endl;
   }
   if (vm.count("use_assimp_export")) {
-#if IS_ASSIMP3
```
Comment (isucan) [inline collada_urdf/src/collada_to_urdf.cpp:600]: Why do we remove this?

Comment (cottsay) [inline collada_urdf/src/collada_to_urdf.cpp:600]: I was hoping to start a conversation about this. I tested assimp2 and assimp3 and they both seem to work fine with this left out.

Any chance @YoheiKakiuchi can shed some light on why the entire export.h was intentionally removed from the assimp2 builds?

Comment (YoheiKakiuchi) [inline collada_urdf/src/collada_to_urdf.cpp:600]: I have no idea why export.h was removed from the assimp2 builds. I think it is no problem if build succeed.

Comment (cottsay) [inline collada_urdf/src/collada_to_urdf.cpp:600]: If it would help, I'd be happy to test this against current Ubuntu versions under groovy/hydro (a groovy backport would be nice). Just let me know which versions I should test and I'll spin up a VM or ten.

Someone should check OSX.

Assuming we find no build problems, do you think that it would be safe to assume that this was a workaround for a "unique" version of assimp on a now unsupported distro?

**Comment by cottsay** [comment]
Okay, I found the reason for leaving export out. Looks like precise uses assimp 2.0.863, which is just before 2.0.885, which is when the export API was added. I just committed code that detects if the export.h is present, and tested it on precise. I also tested on F19, which has assimp 2.0.1071, so we're working with either side of the feature.

I also patched to use VERSION_LESS instead of STRLESS. This was added in cmake 2.8.0 (four years ago) and works on precise. I'll have to modify the other packages to do this comparison as so.

In summary:
- Works 2.0.863 (precise)
- Works 2.0.1071 (F19)
- Works 3.X.X (F20, quantal, etc)

Thanks for your patience on this one.

**Comment by wjwwood** [comment]
I'll test this today as well on OS X.

**Comment by isucan** [comment]
I'll wait for @wjwwood , but otherwise +1 from me.

**Comment by wjwwood** [comment]
Thanks @isucan, I am testing it right now.

**Comment by wjwwood** [comment]
Building it against `assimp 3.0.1270` succeeded, building it against `assimp 2.0.863` now.

**Comment by wjwwood** [comment]
@isucan what would be a good thing to run which would exercise this code? rviz?

**Comment by wjwwood** [comment]
More specifically something that loads collada file(s) from a urdf file?

**Comment by isucan** [comment]
rviz would be good. do you mean a urdf from a collada file?

**Comment by wjwwood** [comment]
No, but I assumed that this was used to load collada files which were referenced as geometry in urdf files, but if there are other better examples I can try those.

**Comment by cottsay** [comment]
Thanks for looking into this. If you need me to squash commits or anything else, please let me know.

**Comment by wjwwood** [comment]
Ok, both built, but now I am going to build rviz on top and test that set of patches too.

**Comment by wjwwood** [comment]
+1 from me, the rviz testing went well.
